### DISCUSSION TITLE
Créer un interfaçage avec des logiciels métiers

### DISCUSSION THREAD

**Discussion by gauthierandre** [betagouv/ma-cantine#1003] [Retours utilisateur·rice]
Lors de rencontres avec plusieurs usagers ces derniers jours (webinaires, One to One), le souhait de simplifier la saisie des données d'approvisionnement sur ma cantine, pour ceux qui ont déjà un logiciel de gestion (easyliss, salamandre etc.), m'a été remonté. Quelques utilisateurs ont des contacts dans ces structures qui sont prêt à discuter avec nous/nous aider sur le sujet. 

Est-ce qu'on peut penser cet interfaçage autrement qu'un export (du logiciel métier) / import (via MaCantine)? Ce qui, pour le coup, ne sera pas nécessairement une simplification ?

**Comment by alemangui** [comment]
Super intéressant. À mon avis il y a plusieurs possibilités. Juste petite question par rapport à 

> Est-ce qu'on peut penser cet interfaçage autrement qu'un export (du logiciel métier) / import (via MaCantine)? Ce qui, pour le coup, ne sera pas nécessairement une simplification ?

Ce ne serait pas une simplification car déjà possible de faire l'import CSV c'est ça ? Ça pourrait être une simplification si par exemple le logiciel a un format spécifique d'import. Dans ce cas, on pourrait faire en sorte que ma cantine puisse lire ces fichiers. La simplification viendrait du fait qu'aucune transformation supplémentaire serait nécessaire. Mais j'ai peut-être mal compris le truc.

**Reply by gauthierandre** [reply]
merci pour ta réponse. 
Effectivement, ça n'est pas aidant cette solution. C'est pour ça, je voulais voir comment vous pouviez l'imaginer au-delà de ça. S'il y a un format d'import spécifique sans transformations supplémentaires, ça serait cool. N'hésitez pas s'il y a besoin de contacts chez Easyliss par exemple, je peux récupérer des coordonnées.

**Comment by gauthierandre** [comment]
CD37 ok pour nous fournir un contact chez EL si besoin.

**Reply by hfroot** [reply]
EL = Easyliss ?
Est-ce que c'est possible de demander un exemple d'un export de données (ou on peut envoyer un mail vers eux nous-même si c'est plus facile) ?

**Reply by gauthierandre** [reply]
oui, désolé pour l'abréviation. Je fais la demande.

**Comment by hfroot** [comment]
C'est quoi notre position en tant qu'équipe vers des autres outils ? Par exemple, si on bosse sur les interfaçage avec quelques outils en particuliers, est-ce qu'on doit éviter donner l'impression qu'on conseil ces outils etc ?

**Reply by gauthierandre** [reply]
La difficulté c'est qu'il faudrait faire pour tous et pas seulement valoriser le schéma d'un seul. C'est le plus long.

**Comment by hfroot** [comment]
Dans ces rencontres, est-ce que vous avez identifiés des freins avec notre outil d'import massif actuellement ?

**Reply by gauthierandre** [reply]
Des freins il y en a : sur le fait d'avoir les données ; l'intérêt de remplir le tableur.
Il y a celle que je t'ai mentionnées sur une autre carte avec les noms en anglais ou pas incomplet.
Tout le monde ne trouve pas la procédure extrêmement clair. La case d'import est vue immédiatement, iels cherchent ensuite les modèles. Pas toujours très évident à ce moment. Est-ce qu'on pourrait valoriser les cases obligatoires sur l'excel à télécharge, ou le csv par des codes couleurs ? je sais que c'est sur la procédure en bas mais elle n'est pas lue..

**Reply by valeriemerle** [reply]
J'ai refait des tests sur l'import massif. Cette procédure me semble trop "experte" pour utilisateurs avertis et compétents en informatique. Les champs des tetes de colonne ne sont pas assez explicites. Il n'est pas clair que les info demandées concernent les cuisines site. Pour l'import, il n'est pas précisé d'enlever la ligne des champs de colonne d'où des erreurs d'import. Si on ne renseigne que le code postal on ne retrouve pas le lieu geographique de la cantine mais celui du bureau de poste principal.
Cette procédure d'import de permet pas de faire le lien entre cuisine centrale et satellites. Lors de l'import on perd l'info sur les cuisines centrales.( elle n'est pas dans la cantine créé).

**Reply by alemangui** [reply]
> Cette procédure d'import de permet pas de faire le lien entre cuisine centrale et satellites.

On a un champ pour spécifier le SIRET de la cuisine centrale non ?

> Pour l'import, il n'est pas précisé d'enlever la ligne des champs de colonne d'où des erreurs d'import.

On peut laisser la ligne d'entête, ça ne produit pas des erreurs. Les erreurs venaient d'autre part ama.

> Si on ne renseigne que le code postal on ne retrouve pas le lieu geographique de la cantine mais celui du bureau de poste principal.

C'est où que tu vois le bureau de poste ?

**Reply by alemangui** [reply]
Par rapport à l'outil étant trop "expert", la cible de cet outil sont des administrateurs qui ont l'habitude de traiter avec des gros fichiers Excel, DB, etc. Quelqu'un.e ayant moins plusieurs douzaines de cantines et qui n'est pas à l'aise avec la gestion de données ne devrait pas utiliser l'import de masse je pense.
